### PR TITLE
fix: Scroll for big charts

### PR DESCRIPTION
**Pull request by maloun96**
### SUMMARY
Add scroll to view entire Time-Series Bar Chart

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE
image
AFTER
Kapture 2020-12-11 at 11 48 57

### TEST PLAN
Add a new chart
Select Time-series Bar Chart

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API
- [x] https://github.com/apache/incubator-superset/issues/11859

### CODE REVIEW

**Comment by maloun96** [comment]
@junlincc @rusackas If you have any idea to improve, I didn't found better solution

**Review by junlincc — APPROVED** [review]
ezgif-1-b5a2958f1158
In dashboard
ezgif-3-8720f6efab6b

LGTM~ @rusackas if the code looks good, we can merge~ Thank you Victor! this is valuable! 


@graceguo-supercat

File: superset-frontend/src/chart/Chart.jsx
Comment (rusackas) [inline superset-frontend/src/chart/Chart.jsx:91]:
```suggestion
    width: auto !important;
```
The above suggestion might also remove the need for `!important`... if that's still not high enough specificity of a selector, we could perhaps play other tricks, like the double ampersand trick, e.g. `&& > svg {`

File: superset-frontend/src/chart/Chart.jsx
Comment (rusackas) [inline superset-frontend/src/chart/Chart.jsx:218]:
```suggestion
            style={{ width, overflow: 'auto' }}
```
does auto work just as well here, so we don't _always_ have a scrollbar?

**Review by rusackas — CHANGES_REQUESTED** [review]
Asked a couple of questions... just want to make sure they're addressed before this gets merged based on Junlin's approval :)

File: superset-frontend/src/chart/Chart.jsx
Comment (maloun96) [reply superset-frontend/src/chart/Chart.jsx:91]: Doesn't work without !important

**Comment by mistercrunch** [comment]
This used to work, unclear what created the regression here, but could be related to the recent work with the data south pane.

I haven't pull the branch here, but I think it's likely that this may bring in regressions around showing scrollbars where they should not be shown (that scroll by a few pixels).

I think typically the scrolling (if needed) should be declared in the visualization plugin itself.

File: superset-frontend/src/chart/Chart.jsx
Comment (mistercrunch) [reply superset-frontend/src/chart/Chart.jsx:90]: Yes we can't assume that there will be SVG or that it should overflow one way or another.

File: superset-frontend/src/chart/Chart.jsx
Comment (mistercrunch) [reply superset-frontend/src/chart/Chart.jsx:218]: many visualization plugins may end up overflowing by a few pixels for whatever reason and we may not want to show the scrollbar.

**Review by mistercrunch — CHANGES_REQUESTED** [review]
I suggest that this should be defined in the plugin itself.

We should identify what broke recently though since I'm certain bar charts used to scroll horizontally

**Comment by junlincc** [comment]
> I haven't pull the branch here, but I think it's likely that this may bring in regressions around showing scrollbars where they should not be shown (that scroll by a few pixels).

tested and looked through if not all, most of the charts, scrollbar only appears when needed, in Bar, Time Series Bar, Table, Pivot Table, Time Series table,Calendar Heatmap etc. not seeing unnecessary scrollbars on chart, yet...

**Comment by rusackas** [comment]
Today there were some bugs that occurred with scrolling charts on the dashboards that needed addressing. The solution was to make things NOT scroll, and mov the responsibility for Filter Box's scrolling to happen in the plugin. I think we should do the same here, so we should use the same solution here, and keep the responsibility for scrolling handled by the plugin itself.

**Comment by rusackas** [comment]
@junlincc @maloun96 Sorry to close the PR. It can be re-opened if you object, of course, but I'm wondering if we could take a fresh look at this, and make changes in the Timeseries chart plugin if that is the one has this specific problem.

**Comment by junlincc** [comment]
Agreed.

**Pull request closed without merging**
